pilot/cmd/pilot-discovery/admission/controller.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package admission
+
+import (
+	"github.com/hashicorp/go-multierror"
+	"istio.io/istio/pilot/platform/kube"
+	"istio.io/istio/pilot/platform/kube/admit"
+)
+
+// NewController creates and runs an instance of the pilot admission controller with the given arguments.
+func NewController(kubeContig string, args admit.ControllerOptions) (*admit.AdmissionController, error) {
```

Comment:
Is kubeContig what we are using to avoid a collision?

Reply:
:) ... fixed the type-o